modules/hotswap/src/main/scala/io/janstenpickle/trace4cats/hotswap/HotswapConstructor.scala

```diff
@@ -0,0 +1,36 @@
+package io.janstenpickle.trace4cats.hotswap
+
+import cats.Applicative
+import cats.effect.kernel.{Resource, Temporal}
+import cats.kernel.Eq
+import cats.syntax.applicative._
+import cats.syntax.eq._
+import cats.syntax.functor._
+import cats.syntax.flatMap._
+
+trait HotswapConstructor[F[_], A, B] {
+  def swap(a: A): F[Boolean]
+  def get: Resource[F, (A, B)]
+  def getA: F[A]
```

Comment:
I don't like `getA`, `getB` names. Because their return types are different and there is no hint of what they mean.